pkg/builder/util.go

```diff
@@ -63,33 +68,104 @@ func DefaultPathEnv(os string) string {
 	return DefaultPathEnvUnix
 }
 
-// parseOutput parses the output string and returns the output type and destination.
-func parseOutput(output string) (string, string, error) {
-	if output == "" {
-		return "", "", nil
+// parseOutput parses --output
+// Refer to https://github.com/moby/buildkit/blob/master/cmd/buildctl/build/output.go#L56
+func parseOutput(exports string) ([]client.ExportEntry, error) {
+	var entries []client.ExportEntry
+	if exports == "" {
+		return entries, nil
 	}
 
-	// Example: type=tar,dest=path
-	matched, err := regexp.Match(`^type=[\s\S]+,dest=[\s\S]+$`, []byte(output))
+	e, err := parseOutputCSV(exports)
```

Comment:
What's csv for?

Reply:
The output `type=tar,image=imagename,push=true` is CSV-format.